alphagov/paas-cf
# [#161387488] Send metrics to Prometheus from paas-metrics

# What

We extended the paas-metrics app to send metrics to Prometheus while still sending the metrics to Datadog.

Changes:
 * Added a /metrics endpoint which exposes the Prometheus metrics for scraping
 * We added a Unit field to the collected metrics, so we can use it as a suffix for the Prom. metrics (e.g. if the Unit is "seconds" and the metric is "foo.bar", then the Prom. metric name will be "paas_foo_bar_seconds")
 * The CF_SKIP_SSL_VALIDATION env var was removed from the app as it's not needed anymore
 * Use Go 1.11

# How to review

1. Deploy your CF pipeline from this branch with DataDog enabled.
  ```
  ENABLE_DATADOG=true BRANCH=paas_metrics_prometheus_161387488 make dev pipelines
  ```

2. Check if the metrics app is serving the metrics:

```
https://paas-metrics.${DEPLOY_ENV}.dev.cloudpipelineapps.digital/metrics
```

3. Check if Prometheus has the new metrics (prefixed with `paas_`)

```
https://prometheus.${DEPLOY_ENV}.dev.cloudpipeline.digital/graph?g0.range_input=1h&g0.expr=paas_aiven_estimated_cost_pounds&g0.tab=0
```

4. Check if these metrics are also available in DataDog for your env.

# Who can review

Not me.
